crates/cgp-field/src/impls/use_field.rs

```diff
@@ -6,17 +14,54 @@ use cgp_type::traits::has_type::ProvideType;
 use crate::traits::has_field::{FieldGetter, HasField};
 use crate::traits::has_field_mut::{HasFieldMut, MutFieldGetter};
 
+/// Type provider for field access.
+///
+/// This type implements the provider pattern for field access, allowing
+/// fields to be accessed through the type system. It uses phantom types
+/// to ensure type safety.
+///
+/// # Type Parameters
+///
+/// * `Tag` - Phantom type identifying the field
+///
+/// # Examples
+///
+/// ```rust,ignore
+/// use cgp_field::UseField;
+/// use core::marker::PhantomData;
+///
+/// struct Name;
+/// let provider = UseField(PhantomData::<Name>);
+/// ```
```

Comment:
The `UseField` pattern is used for implementing getter traits using `HasField`. Something like:

```rust
#[cgp_component {
    provider: NameGetter,
}]
pub trait HasName {
    fn get_name(&self) -> &String;
}

impl<Context, Tag> NameGetter<Context> for UseField<Tag>
where
    Context: HasField<Tag, Value = String>,
{
    fn get_name(context: &Context) -> &String {
        context.get_field(PhantomData)
    }
}

pub type GetNameFromNameField = UseField<symbol!("name")>;
```

We will cover the pattern in later chapters of the book.